### PR TITLE
BuildResidentialHPXML measure: Shared HVAC

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Add arguments:
* `cooling_system_cooling_efficiency_kw_per_ton` (required)
* `shared_hvac_heating_loop_power` (optional)
* `shared_hvac_heating_fan_coil_power` (optional)
* `shared_hvac_heating_water_loop_heat_pump_heating_efficiency` (optional)
* `shared_hvac_cooling_loop_power` (optional)
* `shared_hvac_cooling_fan_coil_power` (optional)
* `shared_hvac_cooling_water_loop_heat_pump_cooling_capacity` (optional)
* `shared_hvac_cooling_water_loop_heat_pump_cooling_efficiency` (optional)
* `shared_hvac_fan_coil_is_ducted` (optional)
* `shared_hvac_heat_pump_shared_loop_power` (optional)

Change arguments:
* `heating_system_type_choices << "Shared #{HPXML::HVACTypeBoiler}"`
* `cooling_system_type_choices << "Shared #{HPXML::HVACTypeChiller}"`
* `cooling_system_type_choices << "Shared #{HPXML::HVACTypeCoolingTower}"`
* `heat_pump_type_choices << "Shared #{HPXML::HVACTypeHeatPumpGroundToAir}"`

## Checklist

Not all may apply:

- [ ] EPvalidator.rb has been updated
- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected regression test changes on CI

### CODE REVIEW

**Comment by joseph-robertson** [comment]
@shorowit Can you give me any guidance on "adding a boolean argument to the reporting measure for assigning shared system energy to the building unit or not"? See https://trello.com/c/SZ61XHud/57-shared-systems for reference.

**Comment by joseph-robertson** [comment]
Seems like, as an alternative, we could add "shared" end use columns to "results_annual.csv" and "results_timeseries.csv".

File: BuildResidentialHPXML/measure.rb [hunk 1]
```diff
@@ -1012,6 +1013,12 @@ def arguments(model)
     arg.setDefaultValue(false)
     args << arg
 
+    arg = OpenStudio::Measure::OSArgument::makeBoolArgument('heating_system_is_shared_system', true)
+    arg.setDisplayName('Heating System: Is Shared System')
+    arg.setDescription('Whether the heating system is a shared system. If true, assumed to serve all the units in the building. This is valid only for a boiler.')
+    arg.setDefaultValue(false)
+    args << arg
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:1020]: I suggest removing this argument and changing `heating_system_type_choices` to split the boiler option into in-unit boiler vs shared boiler. Then you can automatically map IsSharedSystem in the HPXML file.

File: BuildResidentialHPXML/measure.rb [hunk 2]
```diff
@@ -1069,6 +1083,12 @@ def arguments(model)
     arg.setUnits('W/CFM')
     args << arg
 
+    arg = OpenStudio::Measure::OSArgument::makeBoolArgument('cooling_system_is_shared_system', true)
+    arg.setDisplayName('Cooling System: Is Shared System')
+    arg.setDescription('Whether the cooling system is a shared system. If true, assumed to serve all the units in the building. This is valid only for a chiller.')
+    arg.setDefaultValue(false)
+    args << arg
+
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:1091]: This argument is not needed. IsSharedSystem in the HPXML file should always be true for chillers and always false for other cooling system types.

File: BuildResidentialHPXML/measure.rb [hunk 3]
```diff
@@ -1207,6 +1227,65 @@ def arguments(model)
     arg.setUnits('W/CFM')
     args << arg
 
+    arg = OpenStudio::Measure::OSArgument::makeBoolArgument('heat_pump_is_shared_system', true)
+    arg.setDisplayName('Heat Pump: Is Shared System')
+    arg.setDescription('Whether the heat pump is a shared system. If true, assumed to serve all the units in the building. This is valid only for a ground-to-air heat pump.')
+    arg.setDefaultValue(false)
+    args << arg
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:1234]: I suggest removing this argument and changing heat_pump_type_choices to split the GSHP option into GSHP w/ individual loop vs GSHP w/ shared loop. Then you can automatically map IsSharedSystem in the HPXML file.

File: tasks.rb [hunk 4]
```diff
@@ -172,6 +172,20 @@ def create_osws
...
+    # 'base-hvac-shared-boiler-only-baseboard.osw' => 'base.osw',
+    # 'base-hvac-shared-boiler-only-fan-coil.osw' => 'base.osw',
+    # 'base-hvac-shared-boiler-only-fan-coil-ducted.osw' => 'base.osw',
+    # 'base-hvac-shared-boiler-only-water-loop-heat-pump.osw' => 'base.osw',
+    # 'base-hvac-shared-chiller-only-baseboard.osw' => 'base.osw',
+    # 'base-hvac-shared-chiller-only-fan-coil.osw' => 'base.osw',
+    # 'base-hvac-shared-chiller-only-fan-coil-ducted.osw' => 'base.osw',
+    # 'base-hvac-shared-chiller-only-water-loop-heat-pump.osw' => 'base.osw',
+    # 'base-hvac-shared-cooling-tower-only-water-loop-heat-pump.osw' => 'base.osw',
+    # 'base-hvac-shared-ground-loop-ground-to-air-heat-pump.osw' => 'base.osw',
```
Comment (shorowit) [inline tasks.rb:188]: Why are these all commented out? With 'extra-' versions used instead (below)?

File: BuildResidentialHPXML/measure.rb [hunk 5]
```diff
@@ -949,6 +949,7 @@ def arguments(model)
     cooling_system_type_choices << HPXML::HVACTypeRoomAirConditioner
     cooling_system_type_choices << HPXML::HVACTypeEvaporativeCooler
     cooling_system_type_choices << HPXML::HVACTypeMiniSplitAirConditioner
+    cooling_system_type_choices << HPXML::HVACTypeChiller
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:952]: Need to handle Cooling Towers (which would always require a WLHP).

**Review by shorowit — COMMENTED** [review]
Some initial comments below.

Also, I think we need to make it easier to understand how, e.g., fan coils and water loop heat pumps are defined. Not immediately sure what direction to go. Could consider A) adding the distribution type (e.g., fan coil vs WLHP vs baseboard) into the foo_system_type_choices options, or B) Adding a single additional argument for shared systems to describe the distribution type. Pros/cons to both. For example, with option A, would need error/warning-checking if say boiler w/ fan coil but chiller w/ WLHP. Or for example, with option B, would need error/warning-checking if say cooling tower w/ baseboard or cooling tower w/ fan coil

File: tasks.rb (hunk 4 above)
Comment (joseph-robertson) [reply tasks.rb:188]: Because the geometry portions of the built hpxml files have discrepancies with the test xmls...

**Comment by joseph-robertson** [comment]
> @shorowit Can you give me any guidance on "adding a boolean argument to the reporting measure for assigning shared system energy to the building unit or not"? See https://trello.com/c/SZ61XHud/57-shared-systems for reference.

@shorowit Did you see this question?

**Pull request closed without merging**